src/data-sources/DataSourceList.tsx

```diff
@@ -1,122 +1,163 @@
 import { useDataSources } from '@/data-sources/hooks/useDataSources';
-import { DataSourceConfig, DataSourceType } from '@/data-sources/types';
+import { DataSourceConfig } from '@/data-sources/types';
 import { useSettingsContext } from '@/settings/hooks/useSettingsNav';
+import AirtableIcon from '@/settings/icons/airtable';
+import GoogleSheetsIcon from '@/settings/icons/google-sheets';
+import ShopifyIcon from '@/settings/icons/shopify';
 import { slugToTitleCase } from '@/utils/string';
 
+import './data-source-list.scss';
+
 const DataSourceList = () => {
 	const { dataSources, loadingDataSources, deleteDataSource, fetchDataSources } = useDataSources();
-	const [ deleteDialogOpen, setDeleteDialogOpen ] = useState( false );
+	const [ dataSourceToDelete, setDataSourceToDelete ] = useState< DataSourceConfig | null >( null );
 	const { pushState } = useSettingsContext();
 
-	const onCancelDeleteDialog = ( event: DialogInputEvent ) => {
-		if ( event?.type === 'click' ) {
-			setDeleteDialogOpen( false );
-		}
+	const onCancelDeleteDialog = () => {
+		setDataSourceToDelete( null );
 	};
 
-	const openDeleteDialog = () => setDeleteDialogOpen( true );
-
-	const onServiceTypeSelected = ( serviceToAdd: DataSourceType ) => {
-		const newUrl = new URL( window.location.href );
-		newUrl.searchParams.set( 'addDataSource', serviceToAdd );
-		pushState( newUrl );
-	};
+	const onDeleteDataSource = ( source: DataSourceConfig ) => setDataSourceToDelete( source );
 
-	const onEditClick = ( uuidToEdit: string ) => {
+	const onEditDataSource = ( uuidToEdit: string ) => {
 		const newUrl = new URL( window.location.href );
 		newUrl.searchParams.set( 'editDataSource', uuidToEdit );
 		pushState( newUrl );
 	};
 
-	const onDeleteConfirm = async ( uuid: string ) => {
-		await deleteDataSource( uuid ).catch( () => null );
-		setDeleteDialogOpen( false );
+	const onConfirmDeleteDataSource = async ( source: DataSourceConfig ) => {
+		await deleteDataSource( source ).catch( () => null );
+		setDataSourceToDelete( null );
 		await fetchDataSources().catch( () => null );
 	};
 
-	if ( loadingDataSources ) {
-		return (
-			<>
-				{ __( 'Loading data sources...', 'remote-data-blocks' ) }
-				<Spinner />
-			</>
-		);
-	}
-
 	const getValidDataSources = () => {
 		return dataSources.filter( source => SUPPORTED_SERVICES.includes( source.service ) );
 	};
 
 	const renderDataSourceMeta = ( source: DataSourceConfig ) => {
-		if ( source.service === 'airtable' ) {
-			return (
-				<>
-					<Tag id="airtable-base" label="Base" value={ source.base.name } />
-					<Tag id="airtable-table" label="Table" value={ source.table.name } />
-				</>
-			);
+		const tags = [];
+		switch ( source.service ) {
+			case 'airtable':
+				tags.push( source.base.name, source.table.name );
+				break;
+			case 'shopify':
+				tags.push( source.store );
+				break;
+			case 'google-sheets':
+				tags.push( source.spreadsheet.name, source.sheet.name );
+				break;
+		}
```

Comment:
New designs omit the key/label name for the tags. I prefer having them as it makes it clear what those values are for - e.g. it makes it clear which is the spreadsheet name and the sheet name for Google Sheet. Would it be possible to update the designs to include that if we agree on it ?